core/src/main/java/org/mule/component/DefaultComponentLifecycleAdapter.java

```diff
@@ -339,7 +338,7 @@ public Object invoke(MuleEvent event) throws MuleException
         {
             if (componentObject == null)
             {
-                throw new ComponentException(createStaticMessage("componentObject is null"), RequestContext.getEvent(), component);
+                throw new ComponentException(createStaticMessage("componentObject is null"), event, component);
```

Comment:
The MessageProcessorNotificationExecutionInterceptor updates the RC with the eventToProcess that in this case will be the one passed to the ComponentException here.